test/e2e/app-dir/app-static/app-static-set-tag-stale.test.ts

```diff
@@ -0,0 +1,138 @@
+import { nextTestSetup } from 'e2e-utils'
+import { retry } from 'next-test-utils'
+
+describe('app-dir set-tag-stale', () => {
+  const { next, skipped, isNextDev } = nextTestSetup({
+    files: __dirname,
+    skipDeployment: true,
+  })
+
+  if (skipped || isNextDev) {
+    it('should skip test', () => {})
+    return
+  }
+
+  it('should mark single tag as stale with setTagStale', async () => {
+    // First, get the initial page
+    const $ = await next.render$('/stale-tag-test')
+    const initialRandom = $('#random').text()
+    const initialNow = $('#now').text()
+
+    // Mark tag as stale
+    await next.fetch('/api/set-tag-stale-node?tag=stale-test')
+
+    // The page should still show the same content initially (not immediately invalidated)
+    const $2 = await next.render$('/stale-tag-test')
+    const secondRandom = $2('#random').text()
+    const secondNow = $2('#now').text()
+
+    // Should be the same since it's marked stale but not invalidated yet
+    expect(secondRandom).toBe(initialRandom)
+    expect(secondNow).toBe(initialNow)
```

Comment:
I wonder if we should simulate a slow revalidation to avoid false positives here, since the API is non-blocking